### ISSUE TITLE
Matching Batch-Sizes and Buffer-Sizes

### ISSUE THREAD

**Issue by markif** [zeromq/jeromq#622]
User do not have the option to specify Config.IN_BATCH_SIZE nor Config.OUT_BATCH_SIZE. However, in case the user specifies large Options.rcvbuf or Options.sndbuf (in out case the payload of one message is ~16MB) it could be worthwhile to adjust Config.IN_BATCH_SIZE or  Config.OUT_BATCH_SIZE accordingly in order to reduce system calls.

**Comment by fredoboulo** [comment]
Maybe another options could be created to de-correlate the batch sizes and rcvbuf/sndbuf. 
Did you get a performance improvement with your approach?

**Comment by markif** [comment]
We experienced improvements for setups with large payloads allowing us to better utilize the troughput of networks >=10Gbit/s. I did this change quite a while ago (this is the reason why I cannot fully recall the details – in production we use a patched 0.4.0 version) and thought it could make sense to integrate it into the latest version. Given I remember things right the main difference was way less system/OS calls.
